packages/database/test/exp/integration.test.ts

```diff
@@ -95,14 +105,113 @@ describe('Database@exp Tests', () => {
     const [snap1, snap2] = await ea.promise;
     expect(snap1).to.equal('a');
     expect(snap2).to.equal('b');
+    unsubscribe();
+  });
+
+  // Tests to make sure onValue's data does not get mutated after calling get
+  it('calls onValue only once after get request with a non-default query', async () => {
+    const db = getDatabase(defaultApp);
+    const { ref: testRef } = getUniqueRef(db);
+    const initial = [{ name: 'child1' }, { name: 'child2' }];
+    const ec = EventAccumulatorFactory.waitsForExactCount(1);
+
+    await set(testRef, initial);
+    const unsubscribe = onValue(testRef, snapshot => {
+      ec.addEvent(snapshot.val());
+    });
+    await get(query(testRef, limitToFirst(1)));
```

Comment:
Could you turn this into a table test and apply this test case to a few more interesting queries?

```
const queries = [
  query(testRef, limitToFirst(1)),
  query(testRef, startAt("child2")),
  ... etc
]

queries foreach { q => runTest(q) }
```

Some inspiration from the query integration tests: https://github.com/firebase/firebase-js-sdk/blob/master/packages/database-compat/test/query.test.ts#L38.

Reply:
Will do.